pkg/aks/create.go

```diff
@@ -23,17 +23,134 @@ func CreateResourceGroup(ctx context.Context, groupsClient *resources.GroupsClie
+	spec *aksv1.AKSClusterConfigSpec, phase string) error {
+
+	// Create a new managed cluster from the AKS cluster config
+	managedCluster, err := CreateManagedCluster(ctx, cred, spec, phase)
+	if err != nil {
+		return err
+	}
+
+	// Pull the upstream cluster state. If the upstream cluster couldn't be found (not expected), then create a new
+	// cluster. If there was a bad error, return it.
+	upstreamCluster, err := clusterClient.Get(ctx, spec.ResourceGroup, spec.ClusterName)
+	if err != nil {
+		if strings.Contains(err.Error(), spec.ClusterName + " was not found") {
+			return CreateCluster(ctx, cred, clusterClient, spec, phase)
+		} else {
+			return err
+		}
+	}
+
+	// Upstream cluster state was successfully pulled. Merge in updates without overwriting upstream fields: we never
+	// want to overwrite preconfigured values in Azure with nil values. So if a field on the managed cluster is nil,
+	// set it to the upstream value.
+
+	// Cluster id, name, and type are read-only in AKS; location is also not modifiable once the cluster is fully
+	// provisioned; skip
+
+	// Network profile
+	if managedCluster.NetworkProfile == nil {
+		managedCluster.NetworkProfile = upstreamCluster.NetworkProfile
+	} else {
+		// network plugin will never be nil
+		np := managedCluster.NetworkProfile
+		if len(np.NetworkPolicy) == 0 {
+			np.NetworkPolicy = upstreamCluster.NetworkProfile.NetworkPolicy
+		}
+		if len(np.NetworkMode) == 0 {
+			managedCluster.NetworkProfile.NetworkMode = upstreamCluster.NetworkProfile.NetworkMode
+		}
+		if np.DNSServiceIP == nil {
+			managedCluster.NetworkProfile.DNSServiceIP = upstreamCluster.NetworkProfile.DNSServiceIP
+		}
+		if np.DockerBridgeCidr == nil {
+			managedCluster.NetworkProfile.DockerBridgeCidr = upstreamCluster.NetworkProfile.DockerBridgeCidr
+		}
+		if np.PodCidr == nil {
+			managedCluster.NetworkProfile.PodCidr = upstreamCluster.NetworkProfile.PodCidr
+		}
+		if np.ServiceCidr == nil {
+			managedCluster.NetworkProfile.ServiceCidr = upstreamCluster.NetworkProfile.ServiceCidr
+		}
+		if len(np.OutboundType) == 0 {
+			managedCluster.NetworkProfile.OutboundType = upstreamCluster.NetworkProfile.OutboundType
+		}
+		if len(np.LoadBalancerSku) == 0 {
+			managedCluster.NetworkProfile.LoadBalancerSku = upstreamCluster.NetworkProfile.LoadBalancerSku
+		}
+		// not configurable in Rancher so there won't be subfield conflicts. If it's set in Azure, just add it to the
+		// managed cluster obj
+		if np.LoadBalancerProfile == nil {
+			managedCluster.NetworkProfile.LoadBalancerProfile = upstreamCluster.NetworkProfile.LoadBalancerProfile
+		}
+
+		// todo: use the AgentPoolProfiles iterator to pull agent pool changes from upstream before an update
+		//if managedCluster.AgentPoolProfiles == nil {
+		//	for profile := range managedCluster.AgentPoolProfiles {
+		//
+		//	}
+		//}
+
+		// linux profile
+		if managedCluster.LinuxProfile == nil {
```

Comment:
Don't we actually support `LinuxProfile` on Rancher side? So overwriting it from data we got from AKS (upstream) cluster seems we're giving precedence to it in the function that is supposed to propagate Rancher updates to AKS.

Again, goes in line with previous comment about handling fields we support in Rancher instead of handling the fields we don't support.